source/common/res/features/budget-to-goal/main.js

```diff
@@ -0,0 +1,163 @@
+            }
+          });
+        },
+
+        observe(changedNodes) {
+          console.log(changedNodes);
+          if (changedNodes.has('budget-inspector-goals')) {
+            // In theory, if there's a 'budget-inspector-goals' classed node, then the user is or has changed the goal.
+            // However, there seems to be some disconnect between this event and when YNAB updates the values.
+            setTimeout(function () {
+              ynabToolKit.budgetBalanceToGoal.invoke();
+            }, 60000);
+          } else if (changedNodes.has('budget-inspector')) {
+            // User has changed selected budget item or changed budgeted amount.
+            ynabToolKit.budgetBalanceToGoal.invoke();
+          }
+        },
+
+        addBudgetVersionIdObserver() {
+          let applicationController = ynabToolKit.shared.containerLookup('controller:application');
+          applicationController.addObserver('budgetVersionId', function () {
+            Ember.run.scheduleOnce('afterRender', this, resetBudgetView);
+          });
+
+          function resetBudgetView() {
+            ynabToolKit.budgetBalanceToGoal.budgetView = null;
+          }
+        },
+
+        getCategories() {
+          // After using Budget Quick Switch, budgetView needs to be reset to the new budget. The try catch construct is necessary
+          // because this function can be called several times during the budget switch process.
+          if (ynabToolKit.budgetBalanceToGoal.budgetView === null) {
+            try {
+              ynabToolKit.budgetBalanceToGoal.budgetView = ynab.YNABSharedLib.getBudgetViewModel_AllBudgetMonthsViewModel()._result;
+            } catch (e) {
+              return;
+            }
+          }
+
+          var categories = [];
+          var masterCats = ynabToolKit.budgetBalanceToGoal.budgetView.categoriesViewModel.masterCategoriesCollection._internalDataArray;
+          var masterCategories = [];
+
+          masterCats.forEach(function (c) {
+            // Filter out "special" categories
+            if (c.internalName === null) {
+              masterCategories.push(c.entityId);
+            }
+          });
+
+          masterCategories.forEach(function (c) {
+            var accounts = ynabToolKit.budgetBalanceToGoal.budgetView
+              .categoriesViewModel.subCategoriesCollection
+              .findItemsByMasterCategoryId(c);
+
+            Array.prototype.push.apply(categories, accounts);
+          });
+
+          return categories;
+        },
+
+        updateInspectorButton(f) {
+          var goalAmount = ynabToolKit.budgetBalanceToGoal.getBudgetGoalLeft(f);
+          var fGoal = ynabToolKit.shared.formatCurrency(goalAmount);
+
+          if (($('.toolkit-balance-to-goal').length) || (goalAmount === '-0')) {
+            return;
+          }
+
+          /* check for positive amounts */
+          var positive = '';
+          if (ynab.unformat(goalAmount) > 0) { positive = '+'; }
```

Comment:
I'd get rid of the extra curlies here, so:

```
if (ynab.unformat(goalAmount) > 0) positive = '+';
```

Or if you want to really get amongst it:

```
const positive = ynab.unformat(goalAmount) > 0 ? '+' : '';
```